cmd/cli/root.go

```diff
@@ -81,8 +89,14 @@ func NewRootCmd() *cobra.Command {
 			ctx.Value(util.SystemManagerKey).(*system.CleanupManager).Cleanup(ctx)
 		},
 	}
-
-	RootCmd.PersistentFlags().String("repo", "", "path to bacalhau repo")
+	// ensure the `repo` key always gets a usable default value, warn if it's not.
+	defaultRepo, err := defaultRepo()
+	if err != nil {
+		RootCmd.Printf("WARNING: %s\n"+
+			"cannot determine default repo location: "+
+			"BACALHAU_DIR or --repo must be set to initialize a node.\n\n", err)
+	}
```

Comment:
> I don't like that we are logging a warning when trying to set default value for a repo even when a user is not relying on the default value as is passing a value for --repo flag

I'm with Walid, I don't think we need to emit this if the user is being explicit.

Reply:
This warning exists for the (rare) case when a user doesn't have `BACALHAU_DIR`, `HOME`, `FIL_WALLET_ADDRESS` + `ROOT_DIR`, or `XDG_CONFIG_HOME` set and doesn't specify a repo path via `--repo`. Without this warning and in the event this case is true users would only see the message:
 `failed to read --repo value: value not found for repo`.